job-master.yaml

```diff
@@ -8,11 +8,14 @@ spec:
     spec:
       hostPID: true
       nodeSelector:
```

Comment:
You can use `nodeAffinity` instead of `nodeSelector` for backward compatibility for K8s 1.23.x and below:

```
spec:
  affinity:
    nodeAffinity: 
      requiredDuringSchedulingIgnoredDuringExecution: 
        nodeSelectorTerms:
        - matchExpressions:
          - key: node-role.kubernetes.io/control-plane
            operator: Exists
        - matchExpressions:
          - key: node-role.kubernetes.io/master
            operator: Exists
```

Reply:
Excellent! I was able to verify these changes with K8s 1.24.x